### PR TITLE
Add tests for (ARM) macOS

### PR DESCRIPTION
**Pull request by adamltyson**

### CODE REVIEW

**Comment by adamltyson** [comment]
Leaving this as draft because I don't think the macOS tests on GH actions are actually testing anything.

**Comment by adamltyson** [comment]
Would someone @brainglobe/swc-neuroinformatics take a look at this, particularly this bit? https://github.com/brainglobe/brainreg/blob/b2395e70a23abd56793a4aa3d0976312fb13a993/tests/tests/test_integration/test_registration.py#L12-L18

`NiftyReg` just seems to work slightly differently across operating systems, which is why we have this huge set of test data. I haven't found a way to handle this gracefully, but I'm curious if anyone has a better idea before we merge this (as it's getting more and more complex).

**Comment by niksirbi** [comment]
I tested this branch on my M2 Mac locally and got lots of errors. See attached files for pytest logs.
[pytest_M2_log.txt](https://github.com/brainglobe/brainreg/files/9859873/pytest_M2_log.txt)

**Comment by adamltyson** [comment]
The error seems to be a missing `libpng`. Could you try if installing this from conda fixes the issue? If so, then this issue is ok with appropriate docs (or conda dependencies).

```
E           Process failed:
E            dyld[79498]: Library not loaded: '/usr/local/opt/libpng/lib/libpng16.16.dylib'
E             Referenced from: '/Users/nsirmpilatze/Code/brainreg/brainreg/bin/nifty_reg/osX/reg_aladin'
E             Reason: tried: '/usr/local/opt/libpng/lib/libpng16.16.dylib' (no such file), '/usr
```

**Comment by niksirbi** [comment]
Sorry, was trying to run it inside the wrong conda environment. I ran it now again and all tests passed!

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -19,8 +19,10 @@ jobs:
         os: [ubuntu-latest]
         # Include one windows run
```
Comment (niksirbi) [inline .github/workflows/test_and_deploy.yml:20]: Nitpicking, but comment should also mention MacOS now

**Review by niksirbi — DISMISSED** [review]
Tests now pass on MacOS ARM.

**Pull request merged**